text/0141-linking-traces.md

```diff
@@ -0,0 +1,447 @@
+
+### Implementation plan: 
+
+For an MVP and to validate that trace links work, we'll introduce span links on today's transaction events. In a future step, we'll ensure that this also works on our EAP storage system.
+
+The rough MVP implementation order:
+
+1. Ensure Storage can store along span links in `transaction` events today. If not, make necessary changes. Create index/lookup table so that we can query the "next" trace of a given root span/transaction.
+2. Ensure Relay can forward `transaction` events with span links. If not, make necessary changes. Specify types, expected payload structure and discard malformed links.
+3. Implement PoC in SDK, most probably the JS SDK, to send `transaction` envelopes with span links for previous traces
+4. Add previous and next trace buttons to trace view.
+
+In this order, 1 and 2 have potential overlap. 3 and 4 can happen in parallel, depending on prioritization, available data and capacity.  
+
+Once trace chains consisting of previous and next traces can be shown in the UI based on `transaction` event envelopes, we'll move forward with adding general span link and `previous_trace` span link support to EAP.
+
+## [Alternative] Previous Trace Id
+
+A slightly simpler but less powerful approach to span links would be to specifically store a "previous trace id" attribute on the current root span and serialize it as such in the event envelope item.
+Setting such a previous trace id could happen on `startSpan()` calls or by calling `span.setAttribute()` at any time. 
+
+While this might seem simpler at first glance, it has a some drawbacks:
+- Spans can only have a 1:1 relationship, either with a `traceId` or a `traceId-spanId` value. Span links support a 1:N relationship, where one span can link to multiple other spans in- or outside its trace
+- Span links are an established concept in OpenTelemetry. With Sentry increasingly adapting Otel standards, as well as creating an OTLP endpoint in the future, we should embrace Otel-suggested solutions rather than building our own.
+- Attributes are not indexed in our current data storage solution, meaning queries like getting the "next" instead of previous traces are likely impossible or (best case) require the same level of indexing as span links require to show next traces.
+- POtel SDKs, as well as the browser JS SDKs, already expose span link APIs but they currently no-op and are not sent to Sentry. This is currently very opaque behavior and potentially confusing to users familiar with Otel APIs concepts.
+
+# Answered Questions
+
+- **Can we do this with our current event storage? Are we blocked on EAP?**
+
+  We will certainly need to adapt our current event storage but it is possible to do this without being blocked on EAP. We'll separately also need to adjust the planned span schema for EAP but this is not a blocker but a follow-up task after the MVP implementation.
```

Comment:
any concern to NOT do something like this now? ie if we wait and after EAP is delivered, will there be a lot of complexity to fit in a proper linked traces solution?